include/openspace/interaction/joystickcamerastates.h

```diff
@@ -70,49 +75,77 @@ class JoystickCameraStates : public CameraInteractionStates {
         bool isSticky = false, double sensitivity = 0.0
     );
 
-    AxisInformation axisMapping(int axis) const;
+    void setAxisMappingProperty(const std::string& joystickName, int axis,
+        const std::string& propertyUri, float min = 0.f, float max = 1.f,
+        AxisInvert shouldInvert = AxisInvert::No, bool isRemote = true
+    );
 
-    void setDeadzone(int axis, float deadzone);
-    float deadzone(int axis) const;
+    AxisInformation axisMapping(const std::string& joystickName, int axis) const;
 
+    void setDeadzone(const std::string& joystickName, int axis, float deadzone);
+    float deadzone(const std::string& joystickName, int axis) const;
 
-    void bindButtonCommand(int button, std::string command, JoystickAction action,
-        ButtonCommandRemote remote, std::string documentation);
-    void clearButtonCommand(int button);
-    std::vector<std::string> buttonCommand(int button) const;
+    void bindButtonCommand(const std::string& joystickName, int button,
+        std::string command, JoystickAction action, ButtonCommandRemote remote,
+        std::string documentation);
+    void clearButtonCommand(const std::string& joystickName, int button);
+    std::vector<std::string> buttonCommand(const std::string& joystickName,
+        int button) const;
 
 private:
-    // We use an array for the axes and a map for the buttons since the axis are going to
-    // be accessed much more often and thus have to be more efficient. And storing a few
-    // extra AxisInformation that are not used will not matter that much; finding an axis
-    // location in a potential map each frame, however, would
-
-    std::array<AxisInformation, JoystickInputState::MaxAxes> _axisMapping;
-
-    // This array is used to store the old axis values from the previous frame,
-    // it is used to calculate the difference in the values in the case of a sticky axis
-    std::array<float, JoystickInputState::MaxAxes> _prevAxisValues;
-
-    struct ButtonInformation {
-        std::string command;
-        JoystickAction action;
-        ButtonCommandRemote synchronization;
-        std::string documentation;
+    struct JoystickCameraState {
+        std::string joystickName;
+
+        // We use an array for the axes and a map for the buttons since the axis are going to
+        // be accessed much more often and thus have to be more efficient. And storing a few
+        // extra AxisInformation that are not used will not matter that much; finding an axis
+        // location in a potential map each frame, however, would
+
+        std::array<AxisInformation, JoystickInputState::MaxAxes> axisMapping;
+
+        // This array is used to store the old axis values from the previous frame,
+        // it is used to calculate the difference in the values in the case of a sticky axis
+        std::array<float, JoystickInputState::MaxAxes> prevAxisValues;
+
+        struct ButtonInformation {
+            std::string command;
```

Comment:
Some one-line descriptions here would be nice for the components of the ButtonInformation struct